### PR TITLE
NO-ISSUE: Add retries when waiting for worker agent 

### PR DESCRIPTION
**Pull request by flaper87**
On BM agents can take more than 120s to come up. This change will help us wait in loops of 1 minute for, at most, 10 mins

Backport for #480 

Signed-off-by: Flavio Percoco <flavio@redhat.com>
Co-authored-by: Flavio Percoco <flavio@redhat.com>
(cherry picked from commit bf105c9f40b65e794894d90782ffcb7503ae445f)

### CODE REVIEW

**Review by alknopfler — APPROVED** [review]
LGTM tested

File: deploy-worker/render_worker.sh [hunk 1]
```diff
@@ -193,13 +193,17 @@ index=0
 
 for EDGE in ${ALLEDGECLUSTERS}; do
     create_worker_definitions ${EDGE} ${index}
-    WORKER_AGENT=$(oc --kubeconfig=${KUBECONFIG_HUB} get agent -n ${EDGE} --no-headers | grep worker | awk '{print $1}')
-    if [ "x${WORKER_AGENT}" == "x" ]; then
-        echo "Worker agent is empty for ${EDGE}, sleeping a bit more... 120 secs"
-        sleep 120
+    for a in {1..10}; do
```
Comment (eifrach) [inline deploy-worker/render_worker.sh:196]:
```suggestion
    echo "INFO: Waiting for Worker agent ( max 60 minutes) "
    for a in {1..10}; do
```

File: deploy-worker/render_worker.sh [hunk 2]
```diff
@@ -193,13 +193,17 @@ index=0
...
         WORKER_AGENT=$(oc --kubeconfig=${KUBECONFIG_HUB} get agent -n ${EDGE} --no-headers | grep worker | awk '{print $1}')
+        if [ "x${WORKER_AGENT}" == "x" ]; then
+            echo "Worker agent is empty for ${EDGE}, sleeping a bit more... 60 secs"
+            sleep 60
+            continue
+        fi
```
Comment (eifrach) [inline deploy-worker/render_worker.sh:202]: missing error on timeout

**Review by eifrach — CHANGES_REQUESTED** [review]
some small nits

File: deploy-worker/render_worker.sh (hunk 1 above)
Comment (flaper87) [reply deploy-worker/render_worker.sh:196]: this is a backport. Let's fix it in main and then backport it

**Pull request merged**